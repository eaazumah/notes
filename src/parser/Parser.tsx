import React, {ReactNode} from 'react';
import ParsedText from 'react-native-parsed-text';
import styled from 'styled-components/native';
import {parsers} from './helpers/_constants';

interface Props {
  opacity?: number;
  children: ReactNode;
  numberOfLines?: number;
}

const Parser: React.FC<Props> = props => {
  return (
    <Text
      parse={parsers}
      opacity={props.opacity}
      numberOfLines={props.numberOfLines}>
      {props.children}
    </Text>
  );
};

const Text = styled(ParsedText)<{opacity?: number}>`
  font-size: 15px;
  opacity: ${props => props.opacity || 1};
  color: ${props => props.theme.primaryText};
`;

export default Parser;
