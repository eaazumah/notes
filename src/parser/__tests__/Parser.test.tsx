import {render} from '@testing-library/react-native';
import React from 'react';
import TestProvider from '../../__testUtils__/Providers';
import Parser from '../Parser';

describe('Parser', () => {
  test('Parser renders correctly', () => {
    const text = 'My note';
    const screen = render(
      <TestProvider>
        <Parser text={text} maxHeight={100} />
      </TestProvider>,
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
