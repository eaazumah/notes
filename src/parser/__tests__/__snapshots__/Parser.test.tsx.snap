// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser Parser renders correctly 1`] = `
<HTMLView
  maxHeight={100}
  rootComponentProps={
    Object {
      "maxHeight": 100,
      "overflow": "hidden",
    }
  }
  style={
    Array [
      Object {},
    ]
  }
  textComponentProps={
    Object {
      "style": Object {
        "color": "#242A37",
      },
    }
  }
  value="My note"
/>
`;
